src/PSRule.Rules.Azure/rules/Azure.APIM.Rule.ps1

```diff
@@ -244,6 +244,19 @@ Rule 'Azure.APIM.AvailabilityZone' -Ref 'AZR-000052' -Type 'Microsoft.ApiManagem
     }
 } -Configure @{ AZURE_APIM_ADDITIONAL_REGION_AVAILABILITY_ZONE_LIST = @() }
 
+# Synopsis: API Management instances with API versions prior to 2021-08-01 must update.
+Rule 'Azure.APIM.MinAPIVersion' -Ref 'AZR-000320' -Type 'Microsoft.ApiManagement/service' -Tag @{ release = 'GA'; ruleSet = '2022_12'; } {
+    [Datetime]$version = $TargetObject.apiVersion -replace '-preview', ''
+    [Datetime]$minimumVersion = '2021-08-01'
+
+    if ($version -ge $minimumVersion) {
+        $Assert.Pass()
+    }
+    else {
+        $Assert.Fail().Reason($LocalizedData.APIMVersionDeprecated, $PSRule.TargetName, $TargetObject.apiVersion)
+    }
```

Comment:
This can be simplified to just, the following. However my understand is that it is actually more to do with a different property `properties.apiVersionConstraint.minApiVersion` which is an optional property. 

I would propose some updates to the rule:

- If `apiVersion` exists then it should be greater than or equal to the minimum. If it don't exist then it must be an in-flight resource so it using the latest resource provider anyway.
- If `properties.apiVersionConstraint.minApiVersion` does not exist, fail.
- If `properties.apiVersionConstraint.minApiVersion` exists then it must be greater or equal to the minimum.
- Lets configure the minimum version as a configuration parameter. A lot of features such as soft-delete and others are based on the API version used to call rest endpoints. So we can expect that the minimum version may need to be updated in the future or customers may want to set a minimum to ensure features are available. This may require updating the docs and reasons slighting to still call out that a minimum version will not be supported but also that the rule failed because the minimum version of X was not met.

https://learn.microsoft.com/en-us/azure/templates/microsoft.apimanagement/service?pivots=deployment-language-bicep#apiversionconstraint

```suggestion
    $Assert.Create($version -ge $minimumVersion, $LocalizedData.APIMVersionDeprecated, $PSRule.TargetName, $TargetObject.apiVersion).PathPrefix('apiVersion')
```

Reply:
Good feedback. Changes will be done.

 Could you please explain me what `configuration parameter` would mean in this particular case? I tried to search the `PSRule` and `PSRule for Azure` documentation, but didn't find anything. Do you have an example?

Reply:
@bengeset96 Sure. Maybe configuration parameter is unclear, but I mean using the `$Configuration` to set a configuration option that has a default value but can be overridden.

Such as `AZURE_POLICY_WAIVER_MAX_EXPIRY` defined by https://github.com/Azure/PSRule.Rules.Azure/blob/183637188e992a57946da5c83882dde890a78c5d/src/PSRule.Rules.Azure/rules/Azure.Policy.Rule.ps1#L33-L35.

Docs for [configuration](https://microsoft.github.io/PSRule/v2/concepts/PSRule/en-US/about_PSRule_Variables/#configuration).